stablehlo/dialect/StablehloOps.cpp

```diff
@@ -1025,6 +1025,63 @@ LogicalResult DotOp::verify() {
   return success();
 }
 
+// Certain enum values have custom printers that include `<` `>`
+// i.e. `<PHILOX>` This method parses the enum value without `<` `>`.
```

Comment:
Seems like this should all be handled by default, can you file a bug upstream (or better: fix the ODS infra for having this work natively?)

Reply:
I asked Jeff and he confirmed to me that this is supposed to be the default but it is explicitly overridden in StableHLO here: https://github.com/openxla/stablehlo/blob/b08a70ced3883994b8c410a6439f7114f4325d85/stablehlo/dialect/StablehloEnums.td#L182

Reply:
I guess this is what you meant in the comment here actually maybe? I didn't get it earlier, but then why do we have this just to "undo" it later?

Reply:
Pasting in a conversation from an offline thread from while back:

> Mehdi Amini
> These should display `#mhlo.rng_algorithm<PHILOX>` instead, I suspect there is a bug for this?
>
> Kevin Gleason
> That was something I changed then reverted. This paste is outdated.
>
> It still says `#mhlo.rng_algorithm<PHILOX>` now, only problem is that when inlined it reads `mhlo.rng_bit_generator %arg0, <PHILOX>` including angle brackets due to the custom assembly format for the enum.
>
> Thinking to either accept that, or make a custom printer/parser for these enum values? Is there a different/better way?

This display is inherited from the MHLO bootstrap. Adding ``"`<` $value `>`"`` causes the generic print to display `#stablehlo.rng_algorithm<PHILOX>` instead of `#stablehlo<comparison_direction LT>`. I'd be happy to undo this and then pretty print without custom machinery. These two `rng_` attrs are the only enum attrs that use this display, all the rest use the style of `comparison_direction`. It is easier and more consistent that way, but I'm not sure if there is any historic reason why we would want `rng_*` to display this way.

Reply:
It looks like the `RngAlgorithm` assemblyFormat was introduced in https://github.com/tensorflow/tensorflow/pull/55752 where the author seems to have run into difficulties with default prettyprinting. `RngDistribution` was added a month later, and I think it just followed whatever `RngAlgorithm` did.

Overall, I think we should try to migrate `RngAlgorithm` and `RngDistribution` to be consistent with other attributes.

Reply:
Removed custom printing and fixed test cases. Will wait on feedback before submit. The integration work required to port this to MHLO as well is not too bad, ~25 test files in the TF repo it looked like.